### ISSUE TITLE
More information about running this

### ISSUE THREAD

**Issue by Huzzzaif** [nsidn98/InforMARL#21]
Hi, I was interested in cloning and running the code on my windows machine , is there any read me file that can guide me? I tried doing it on conda, was able to get all the requirements working, but i get stuck on the wandb login?? I don't know what i am doing wrong, i don't know if i should create a new account or do something else?

python -u onpolicy/scripts/train_mpe.py --use_valuenorm --use_popart --project_name "informarl" --env_name "GraphMPE" --algorithm_name "rmappo" --seed 0 --experiment_name "informarl" --scenario_name "navigation_graph" --num_agents 3 --collision_rew 5 --n_training_threads 1 --n_rollout_threads 128 --num_mini_batch 1 --episode_length 25 --num_env_steps 2000000 --ppo_epoch 10 --use_ReLU --gain 0.01 --lr 7e-4 --critic_lr 7e-4 --user_name "marl" --use_cent_obs "False" --graph_feat_type "relative" --auto_mini_batch_size --target_mini_batch_size 128

Running this on the terminal gives me 3 options on wandb, i have made an account there, but its not linking :

<img width="847" alt="Screenshot 2024-06-26 163539" src="https://github.com/nsidn98/InforMARL/assets/20568320/c0bd31a7-5cc6-4c40-9d29-8b4c1402d4e2">

**Comment by nsidn98** [comment]
Do you have a wandb account? And have you [logged into](https://docs.wandb.ai/ref/cli/wandb-login) wandb in your CLI?

**Comment by Huzzzaif** [comment]
Yes i do , I used the API key to login. Should i create anything on wandb?
![image](https://github.com/nsidn98/InforMARL/assets/20568320/925384a0-0cf9-4db3-b692-ca85e5a758ed)


This is what I am running
python -u onpolicy/scripts/train_mpe.py --use_valuenorm --use_popart --project_name "informarl" --env_name "GraphMPE" --algorithm_name "rmappo" --seed 0 --experiment_name "informarl" --scenario_name "navigation_graph" --num_agents 3 --collision_rew 5 --n_training_threads 1 --n_rollout_threads 128 --num_mini_batch 1 --episode_length 25 --num_env_steps 2000000 --ppo_epoch 10 --use_ReLU --gain 0.01 --lr 7e-4 --critic_lr 7e-4 --user_name "huzaif35" --use_cent_obs "False" --graph_feat_type "relative" --auto_mini_batch_size --target_mini_batch_size 128


I am using :
python : Python 3.10.9

Requirements:
gym==0.10.5
numpy==1.19.4
pyglet==1.5.26
protobuf==3.20.0
PyOpenGL==3.1.5
requests==2.22.0
tensorboard==2.4.0
tensorboard-plugin-wit==1.7.0
tensorboardX==2.1
torch==1.8.1
torch-cluster==1.5.9
torch-geometric==2.0.4
torch-scatter==2.0.7
torch-sparse==0.6.10
torch-spline-conv==1.2.1
torchvision==0.9.1
tqdm==4.44.1
typing==3.6.4
typing-extensions==3.7.4.3
wandb==0.10.31

**Comment by nsidn98** [comment]
Can you try `wandb login --relogin`?

Another option is to set the `--use_wandb` flag to not use wandb and rather log data in tensorboard.

**Comment by Huzzzaif** [comment]
I tried re logging, but i still get the same error. 
I used the flag that you suggested and now no login error, but i am getting a lot of DLL errors and the CMD crashes, trying to see what can be done do you have any idea about this?
these are the errors, there are some memory errors, and some DLL errors does this mean my requirements are not installed correctly?
![image](https://github.com/nsidn98/InforMARL/assets/20568320/b5d61844-b7f8-41a2-8cae-2216c8c5af4c)
![image](https://github.com/nsidn98/InforMARL/assets/20568320/1aa1dc7e-ae83-4301-b73c-fd08836b7f73)

**Comment by nsidn98** [comment]
If you are running it on a local machine without enough compute you can try setting a smaller `--num_rollout_threads` instead of the default 128 used in the command. That could maybe help.

**Comment by Huzzzaif** [comment]
I tried reducing it to 32, I am still getting the memory error. I do have a GPU on my machine, so i don't understand why it is not able to handle. I will try to reduce the rollout threads and the batch size next. Is there anything i can do with CUDA?

**Comment by nsidn98** [comment]
So the GPU is required only for the neural network training. However, the code requires multiprocessing on parallel threads (multiple CPUs) for running multiple parallel environments. Maybe you can start with `--num_rollout_threads=2` and then maybe increase from there if that works.

**Comment by Huzzzaif** [comment]
I was able to run it, apparently the wheel creation was failing for torch, so i installed the separately. Had some minor issues with some deprecated code, and it ran after i fixed it.
This is the command for conda:
python -u onpolicy/scripts/train_mpe.py --use_valuenorm --use_popart --project_name "informarl" --env_name "GraphMPE" --algorithm_name "rmappo" --seed 0 --experiment_name "informarl" --scenario_name "navigation_graph" --num_agents 3 --collision_rew 5 --n_training_threads 1 --num_rollout_threads 2 --num_mini_batch 1 --episode_length 25 --num_env_steps 2000000 --ppo_epoch 10 --use_ReLU --gain 0.01 --lr 7e-4 --critic_lr 7e-4 --user_name "marl" --use_cent_obs "False" --graph_feat_type "relative" --auto_mini_batch_size --target_mini_batch_size 128 --use_wandb False
Threads are 2 and wandb is false.

This is what i got ( just wanted to check if i am in the right direction)
![image](https://github.com/nsidn98/InforMARL/assets/20568320/b2e06496-3afb-49bc-bbba-09f9b9a87170)
![image](https://github.com/nsidn98/InforMARL/assets/20568320/06f91535-30a2-41f4-8514-f0f204e7a222)
and finally :
![image](https://github.com/nsidn98/InforMARL/assets/20568320/c3bb83e1-ef3a-4c27-98c2-0d28d9ea0379)
So the training is taking a very long time, is this normal? I assume this is happening because i am running it on the local machine with lesser threads.

Also what should be my next steps.

**Comment by nsidn98** [comment]
Great! It seems to be working. Yeah, with lower number of threads, it takes quite a bit of time to train. It is pretty normal. Increasing the threads to a larger number makes it faster along with GPUs. 

Our lab is working to make JAX version of this so that we can leverage the speedups given by it to train it faster than what we have now. Please close this issue if you feel that this is resolved.
